src/client/wetty/download.ts

```diff
@@ -34,12 +38,23 @@ function onCompleteFile(bufferCharacters: string): void {
     mimeType = 'text/plain';
     fileExt = 'txt';
   }
-  const fileName = `file-${new Date()
-    .toISOString()
-    .split('.')[0]
-    .replace(/-/g, '')
-    .replace('T', '')
-    .replace(/:/g, '')}${fileExt ? `.${fileExt}` : ''}`;
+  let fileName;
+  try {
+    if (typeof fileNameBase64 === "string") {
+      fileName = window.atob(fileNameBase64);
+    }
+  } catch (err) {
+    // Filename wasn't base64-encoded so let's ignore it
+  }
+
+  if (typeof fileName !== "string") {
```

Comment:
Im not a big fan of typeof checks as it implies the type can change. 

Given the type is `string|undefined` could the check be is `fileName === undefined`